lib/Controller/UsersController.php

```diff
@@ -469,6 +500,213 @@ public function create($username, $password, array $groups= [], $email='') {
+	 * @param $userId
+	 */
+	public function resendToken($userId) {
+		$user = $this->userManager->get($userId);
+
+		if ($user === null) {
+			return new TemplateResponse(
+				'core', 'error', [
+				"errors" => [["error" => "User: $userId does not exist"]]
+			],
+				'guest'
+			);
+		}
+
+		if ($user->getEMailAddress() === null) {
+			return new TemplateResponse(
+				'core', 'error', [
+				"errors" => [["error" => $this->l10n->t('Email address not set for: %s', [$userId])]]
+			],
+				'guest'
+			);
+		}
+
+		$token = $this->secureRandom->generate(21,
+			ISecureRandom::CHAR_DIGITS,
+			ISecureRandom::CHAR_LOWER, ISecureRandom::CHAR_UPPER);
+		$this->config->setUserValue($userId, 'owncloud',
+			'lostpassword', $this->timeFactory->getTime() . ':' . $token);
+
+		// data for the mail template
+		$mailData = [
+			'username' => $userId,
+			'url' => $this->urlGenerator->linkToRouteAbsolute('user_management.Users.setPasswordForm', ['userId' => $userId, 'token' => $token])
+		];
+
+		$mail = new TemplateResponse('user_management', 'new_user/email-html', $mailData, 'blank');
+		$mailContent = $mail->render();
+
+		$mail = new TemplateResponse('user_management', 'new_user/email-plain_text', $mailData, 'blank');
+		$plainTextMailContent = $mail->render();
+
+		$subject = $this->l10n->t('Your %s account was created', [$this->defaults->getName()]);
+
+		try {
+			$message = $this->mailer->createMessage();
+			$message->setTo([$user->getEMailAddress() => $userId]);
+			$message->setSubject($subject);
+			$message->setHtmlBody($mailContent);
+			$message->setPlainBody($plainTextMailContent);
+			$message->setFrom([$this->fromMailAddress => $this->defaults->getName()]);
+			$this->mailer->send($message);
+		} catch (\Exception $e) {
+			$this->log->error("Can't send new user mail to " . $user->getEMailAddress() . ": " . $e->getMessage(), ['app' => 'settings']);
+		}
+
+		return new TemplateResponse(
+			'user_management', 'new_user/tokensendnotify', [], 'guest'
+		);
+	}
+
+	/**
+	 * @PublicPage
+	 * @NoAdminRequired
+	 * @NoSubadminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @param $token
+	 * @param $userId
+	 * @param $password
+	 * @return array|RedirectResponse|TemplateResponse
+	 */
+	public function setPassword($token, $userId, $password) {
+		try {
+			$this->checkPasswordSetToken($token, $userId);
+			$user = $this->userManager->get($userId);
+
+			if (!$user->setPassword($password)) {
+				return new TemplateResponse(
+					'core', 'error', [
+						"errors" => [[
+							"error" =>
+								$this->l10n->t('The password can not be set for user %s. ' .
+									'Contact your admin.', [$userId])]]
+					], 'guest'
+				);
+			}
+
+			\OC_Hook::emit('\OC\Core\LostPassword\Controller\LostController', 'post_passwordReset', ['uid' => $userId, 'password' => $password]);
+			\OC_User::unsetMagicInCookie();
+		} catch (UserTokenException $e) {
+			return new TemplateResponse(
+				'core', 'error', [
+					"errors" => [["error" => $e->getMessage()]]
```

Comment:
This might not be translated

Reply:
Properly fetch the exception code and then translate it.